src/wallet/init.cpp

```diff
@@ -107,7 +107,7 @@ void WalletInit::Construct(NodeContext& node) const
         LogPrintf("Wallet disabled!\n");
         return;
     }
-    auto wallet_client = interfaces::MakeWalletClient(*node.chain, args);
+    auto wallet_client = interfaces::MakeWalletClient(node, *node.chain, args);
```

Comment:
Will there ever be a situation where you would want to pass a  `chain` that differs from the chain referenced by `node`? If not, then this seems to add a potential place for someone to do something incorrectly, compared to something like `interfaces:MakeWalletClient(node, args);`. 

This would also lightly go for the same for `nodeContext` and `chain` in `WalletContext` - `chain` would ultimately just be a shortcut for nodeContext->chain, but that one would be a much more impactful change

Reply:
The idea was to build next to the code as it's a Dogecoin change, but replacing also works. Agreed on the shortcut approach being a bigger change, and keeping in mind how we merge from Bitcoin Core is a consideration, one I'd rather avoid.

I've removed chain from being its own parameter until the final `WalletClientImpl` construction, where it was easier to leave it IMHO.

Reply:
I didn't even take into consideration the merging. Whatever keeps that process simpler is probably the best way 👍